jablib/src/main/java/org/jabref/logic/cleanup/ConvertToBiblatexCleanup.java

```diff
@@ -53,6 +53,31 @@ public List<FieldChange> cleanup(BibEntry entry) {
                 }
             });
         }
+        // If still no 'date' field, try fallback logic
+        if (!entry.hasField(StandardField.DATE)) {
+            changes.addAll(applyDateFallback(entry));
+        }
+
+        return changes;
+    }
+
+    private List<FieldChange> applyDateFallback(BibEntry entry) {
+        List<FieldChange> changes = new ArrayList<>();
+
+        Optional<String> yearValue = entry.getFieldOrAlias(StandardField.YEAR).map(String::trim);
+        if (yearValue.isEmpty()) {
+            return changes;
+        }
+
+        String yearText = yearValue.get();
+        Optional<Date> fallbackDate = Date.parse(yearText);
+        if (fallbackDate.isEmpty()) {
+            return changes;
+        }
+
+        entry.setField(StandardField.DATE, yearText).ifPresent(changes::add);
```

Comment:
Add before as java comment:

    // If year was a full date, move it from year to date field.